### PR TITLE
Fixes #7574, disposals piping construction issues

### LINKED ISSUE #7574: Disposal tubing issues
- ~~De-constructing a disposals bin will temporarily make it look like a regular disposals tube~~ unable to reprojews
  
  until it's unwrenched from the floor
  
- Trying to unwrench an unwelded disposals tube section that is on the same tile as floortiled plating (i.e. firstmost picture) will result in this message: "You can only attach the disposal bin if the floor plating is removed."
  Makes little sense since you're trying to unattach the pipe, not attach it, and it should logically let you unwrench the pipe without having to crowbar the floortile loose anyways.
- The following results in you having a wrenched pipe that you can't unwrench until you get rid of the other pipe first (massive annoyance, illogical, can happen accidentally):
  
  1) Have a bent and straight pipe both wrenched to the underfloor (can happen when trying to replace a straight pipe with a bent pipe, for example)
  2) Weld one of them
  3) Now you can't unwrench the other one, as it tells you "There is already a pipe at that location", even though you're trying to get rid of it, not construct it
  4) To remove the non-welded pipe you have to get rid of the welded one.
- Examining a pipe should tell you if it's wrenched to the underfloor or not.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: code/modules/recycling/disposal-construction.dm [hunk 1]
```diff
@@ -19,7 +19,14 @@
 	var/dpdir = 0	// directions as disposalpipe
 	var/base_state = "pipe-s"
 
-	// update iconstate and dpdir due to dir and type
+/obj/structure/disposalconstruct/examine(mob/user)
+	..()
+	if(anchored)
+		to_chat(user, "<span class='info'>It's bolted down to the floor plating.</span>")
+	else
+		to_chat(user, "<span class='info'>It has been detached from the floor plating.</span>")
```
Comment (PJB3005) [inline code/modules/recycling/disposal-construction.dm]: "Has been" implies that it was attached at some point, which isn't necessarily true.

File: code/modules/recycling/disposal-construction.dm [hunk 2]
```diff
@@ -173,8 +180,22 @@
 			nicetype = "pipe"
 			ispipe = 1
 
+	//This bit is the only one about DETACHING the pipe, so it doesn't need all the checks below.
+	if(anchored && iswrench(I))
```
Comment (PJB3005) [inline code/modules/recycling/disposal-construction.dm]: Could you unify this code block and the one for attacking?

File: code/modules/recycling/disposal.dm [hunk 3]
```diff
@@ -884,27 +884,21 @@
...
-			var/turf/uloc = user.loc
-			var/atom/wloc = W.loc
-			to_chat(user, "Slicing the disposal pipe.")
-			sleep(30)
-			if(!W.isOn())
-				return
-			if(user.loc == uloc && wloc == W.loc)
+			to_chat(user, "You start slicing the disposal pipe.")
+			if(do_after(user, src, 3 SECONDS))
+				if(!src || !W.isOn())
```
Comment (PJB3005) [inline code/modules/recycling/disposal.dm]: `if(src)` can never fail, you need `gcDestroyed`.

File: code/modules/recycling/disposal.dm [hunk 4]
```diff
@@ -1394,17 +1388,11 @@
...
```
Comment (PJB3005) [inline code/modules/recycling/disposal.dm]: Again.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/modules/recycling/disposal-construction.dm (hunk 2 above)
Comment (9600bauds) [reply code/modules/recycling/disposal-construction.dm]: What code for attacking?

Comment (PJB3005) [reply code/modules/recycling/disposal-construction.dm]: I meant attaching*.

**Comment by 9600bauds** [comment]
Tested, works OK

**Review by PJB3005 — APPROVED** [review]

**Comment by PJB3005** [comment]
Alright let's take bets how long this will take to have Travis pass.

I'm gonna bet 30 minutes.

**Pull request merged**